kratos/python/add_properties_to_python.cpp

```diff
@@ -106,6 +106,12 @@ typename Properties::TableType& GetTableHelperFunction1( TContainerType& el,
     return el.GetTable(XVar, YVar);
 }
 
+Properties::Pointer CloneProperties(const Properties& rReference){
+    Properties::Pointer p_new_properties = Kratos::make_shared<Properties>(rReference.Id());
```

Comment:
Why don’t you call clone?

I would say this creates another ref to the same properties rather than cloning it
Have you tested if it actually does not modify the existing one when you modify the new one?

Why don’t you call Clone? Or in case there is no clone I would say you should call the copyCtor

Reply:
There is no clone, and here i am calling the copy constructor (line111). But as it has to be a shared pointer. thats why i can not create the object on the stack directly by the copy constructor.

Reply:
I can of course also make a clone method at the properties object, but i thought this is less intrusive.